core/src/Cuda/Kokkos_Cuda_KernelLaunch.hpp

```diff
@@ -472,13 +502,17 @@ struct CudaParallelLaunchImpl<
       LaunchMechanism>;
 
   inline static void launch_kernel(const DriverType& driver, const dim3& grid,
-                                   const dim3& block, const int shmem,
+                                   const dim3& block, int shmem,
                                    const CudaInternal* cuda_instance,
-                                   const bool prefer_shmem) {
+                                   bool prefer_shmem) {
     if (!Impl::is_empty_launch(grid, block)) {
       Impl::check_shmem_request(cuda_instance, shmem);
       Impl::configure_shmem_preference(base_t::get_kernel_func(), prefer_shmem);
 
+      modify_launch_configuration_if_desired_occupancy_is_specified(
+          driver.get_policy(), cuda_instance->m_deviceProp,
+          get_cuda_func_attributes(), block, shmem, prefer_shmem);
+
```

Comment:
`prefer_shmem` is determined once in the execution of the whole program for each `DriverType`, and yet you're changing it on a per-policy instance basis. Can you explain this (at least with a comment?)

Reply:
And I overlooked that `configure_shmem_preference` does not check whether `prefer_shmem` changed.  Good point.

Reply:
I updated the function.  Please look over the changes.

Reply:
(Minor note: in the future, if you can, please give a SHA of the commit that addresses the comment so I can look at it directly.)

Reply:
It looks like this was in 1da449